src/routing/mod.rs

```diff
@@ -713,39 +728,6 @@ struct FromEmptyRouter<B> {
     request: Request<B>,
 }
 
-/// We need to track whether we're inside an `Or` or not, and only if we then
-/// should we save the request into the response extensions.
-///
-/// This is to work around https://github.com/hyperium/hyper/issues/2621.
-///
-/// Since ours can be nested we have to track the depth to know when we're
-/// leaving the top most `Or`.
-///
-/// Hopefully when https://github.com/hyperium/hyper/issues/2621 is resolved we
-/// can remove this nasty hack.
-#[derive(Debug)]
-struct OrDepth(usize);
```

Comment:
https://github.com/hyperium/hyper/issues/2621 was fixed in h2 0.3.4 which was just published.